skimage/transform/_seam_carving.pyx

```diff
@@ -0,0 +1,217 @@
+                       Py_ssize_t cols):
+    """ For each row, compute the lowest seam value for all its columns.
+
+    This function updates `cumulative_img` such that `cumulative_img[r, c]`
+    is the total energy of the lowest energy seam ending at `(r, c)`.
+
+    Parameters
+    ----------
+    energy_img : (M, N, 1) ndarray
+       The array of cost of removal of each pixel. Seam carving tries to avoid
+       pixels with high costs.
+    cumulative_img : (M, N) ndarray
+       The array to be updated with the total cost of lowest energy seams.
+    track_img : (M, N) ndarray
+        For each pixel, `track_img` stores the relative column offset in
+        the previous row which has the lowest value in `cumulative_img`. This
+        helps in in re-tracing the minimum cost seam.
+    cols : int
+        Number of columns to process.
+    """
+
+    cdef Py_ssize_t r, c, offset, c_idx
+    cdef Py_ssize_t rows = energy_img.shape[0]
+    cdef cnp.double_t min_cost = DBL_MAX
+
+    for c in range(cols):
+        cumulative_img[0, c] = energy_img[0, c, 0]
+
+    for r in range(1, rows):
+        for c in range(cols):
+            min_cost = DBL_MAX
+            for offset in range(-1, 2):
+
+                c_idx = c + offset
+                if (c_idx > cols - 1) or (c_idx < 0):
+                    continue
+
+                if cumulative_img[r-1, c_idx] < min_cost:
+                    min_cost = cumulative_img[r-1, c_idx]
+                    track_img[r, c] = offset
+
+            cumulative_img[r, c] = min_cost + energy_img[r, c, 0]
+
+cdef cnp.uint8_t _mark_seam(cnp.int8_t[:, ::1] track_img,
+                            Py_ssize_t start_index,
+                            cnp.uint8_t[:, ::1] seam_map):
+
+    """ Re-trace the optimal seam from a given column in the last row.
+
+    This function tries to re-track an optimal seam from `start_index` and
+    tries to mark it in `seam_map`. If this seam intersects with any existing
+    seam in `seam_map` the function returns `0` without marking anything. Else
+    it marks the seam in `seam_map` and returns `1`.
+
+    track_img : (M, N) ndarray
+       The array of relative column indices as updated by `_preprocess_image`.
+    start_index : int
+       The column number of the bottom most row from where to start re-tracing
+       the seam.
+    seam_map : (M, N) ndarray
+        The array used to mark seams. If a pixel is marked as as seam it is set
+        to `1`, else `0`.
+
+    Returns
+    -------
+    out : int
+        `1` if seam was marked, `0` is seam inersects and was not marked.
+    """
+    cdef Py_ssize_t rows = track_img.shape[0]
+    cdef Py_ssize_t[::1] current_seam_indices = np.zeros(rows, dtype=np.int)
+    cdef Py_ssize_t row, col
+    cdef cnp.int8_t offset
+    cdef Py_ssize_t seams
+
+    current_seam_indices[rows - 1] = start_index
+    for row in range(rows - 2, -1, -1):
+        col = current_seam_indices[row+1]
```

Comment:
PEP8 is a little tricky around this, but for indexing once I tend to prefer having spaces around operators. `row + 1`